sqlmesh/engines/commands.py

```diff
@@ -114,8 +114,18 @@ def cleanup(
 ) -> None:
     if isinstance(command_payload, str):
         command_payload = CleanupCommandPayload.parse_raw(command_payload)
+
+    expired_schemas: t.Set[str] = set()
     for environment in command_payload.environments:
         evaluator.demote(environment.snapshots, environment.name)
+        expired_schemas.update(
```

Comment:
Do we even need to demote individual snapshots now?

Reply:
Yeah good point. All current engines support CASCADE and I guess if we end up adding one that doesn't support it then we can just add the ability to drop the individual objects within the engine adapter method.